src/Apps/Sell/SellFlowContext.tsx

```diff
@@ -94,10 +101,18 @@ export const SellFlowContextProvider: React.FC<SellFlowContextProviderProps> = (
   }
 
   const goToPreviousStep = () => {
+    trackTappedSubmissionBack(submissionID, state.step)
+
     handlePrev()
   }
 
-  const finishFlow = () => {
+  const finishFlow = async () => {
+    trackConsignmentSubmitted(submissionID)
+
+    await updateSubmission({
+      state: state.isSubmitStep ? "SUBMITTED" : undefined,
+    })
```

Comment:
Instead of setting the state in `updateSubmission`, I am updating it here to avoid setting the state to "SUBMITTED" when clicking the back button on the last step.

Reply:
Sorry, I am confused, `finishFlow` is called only when clicking on the Next button, no?

And in any case I would add a comment in the code, because it is not very obvious

Reply:
Yes, `finishFlow` is only called on the last step when clicking on "Submit Artwork." I've added a comment to it 👍

Reply:
I mean, in `BottomFormNavigation` component (where `finishFlow` is called from), there is a code:

```
isSubmitStep ? actions.finishFlow() : actions.goToNextStep()
```

Which means that `finishFlow` is called only when `isSubmitStep`. Then I don't understand what for this check is: `state: state.isSubmitStep ? "SUBMITTED" : undefined,`, it will always result to "SUBMITTED", no? @olerichter00

Reply:
Yeah, good catch! This check isn't needed anymore; I've removed it 🧹